control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -331,6 +333,43 @@ func (r *HostedControlPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 		meta.SetStatusCondition(&hostedControlPlane.Status.Conditions, newCondition)
 	}
 
+	{
+		r.Log.Info("Reconciling hosted cluster version conditions")
+		newCondition := func() metav1.Condition {
+			timeout, cancel := context.WithTimeout(ctx, 2*time.Second)
+			defer cancel()
+			var clusterVersion configv1.ClusterVersion
+			if err := r.HostedAPICache.Get(timeout, client.ObjectKey{Name: "version"}, &clusterVersion); err != nil {
+				return metav1.Condition{
+					Type:    string(hyperv1.ClusterVersionFailing),
+					Status:  metav1.ConditionUnknown,
+					Reason:  "StatusUnknown",
+					Message: fmt.Sprintf("failed to get clusterversion: %v", err),
+				}
+			}
+			for _, cond := range clusterVersion.Status.Conditions {
+				if cond.Type == "Failing" {
```

Comment:
Besides the failing condition, I think we should surface the Available condition via the Available condition on HostedControlPlane. Right now all it does is mirror the KubeAPIServerAvailable condition

Reply:
I agree we should incorporate CVO's Available condition, but I'm not sure it's so straightforward as moving to use ClusterVersion. Wouldn't we also want to incorporate KubeAPIServerAvailable directly as well since the CVO isn't managing the API server and thus CVO Available doesn't incorporate API server availability?

Perhaps a followup/continuation here after the initial merge?

Reply:
sgtm